Add SnapConfig.unset

gnuoy opened the pull request:
Add SnapConfig.unset
    
Add SnapConfig.unset to allow for config keys to be unset via the
SnapConfig object.

Commit:
Add SnapConfig.unset

Add SnapConfig.unset to allow for config keys to be unset via the
SnapConfig object.

Commit:
Add dotted notation tests

sparkiegeek commented on integration_tests.py:
```diff
@@ -33,3 +34,38 @@ def test_service(snap: Snap, service: str):
+    try:
+        snap.config.unset(['foo', 'goo'])
+    except UnknownConfigKey:
+        pass
```
use `contextlib.suppress` for ease of reading?

```suggestion
    with suppress(UnknownConfigKey):
        snap.config.unset(['foo', 'goo'])
```

sparkiegeek reviewed (changes requested):
Don't think you want the .snap added to the repo?

Commit:
Remove .snap and use cleaner contextlib.suppress

albertodonato commented on integration_tests.py:
```diff
@@ -33,3 +35,36 @@ def test_service(snap: Snap, service: str):
+def test_config_set_unset(snap: Snap):
+    # Unset the test keys before running the test proper just in case
+    # it has been set by another test.
+    with contextlib.suppress(UnknownConfigKey):
```
this shouldn't be needed, since unset won't fail if the config doesn't exist

albertodonato commented on integration_tests.py:
```diff
@@ -33,3 +35,36 @@ def test_service(snap: Snap, service: str):
+    # it has been set by another test.
+    with contextlib.suppress(UnknownConfigKey):
+        snap.config.unset(['foo', 'goo'])
+    # Start Test:
```
this comment could be dropped

albertodonato reviewed (changes requested):
Thanks for the changes! 

Just a couple of nits inline.

Also please run `tox -e format` to ensure formatting, and a unit test for `unset` in `snaphelpers/tests/test_conf.py` (similar to the one for `set`) would also be needed.

Commit:
Remove handling of UnknownConfigKey when unsetting

Commit:
Run format and add unit test

Commit:
Remove unset integration test

The config set and unset require sudo so are not good candidates
for the intergration tests which run unprivileged. Since this
code is covered by the unit tests it seems resonable to remove them.

Commit:
Remove unused import

albertodonato reviewed (approved):
Thanks for fixing CI issues!

I'll take a look at running integration tests as root separately, so that's possible to add tests like the one you had.
